exchanges/gemini/gemini_websocket.go

```diff
@@ -261,15 +317,61 @@ func (g *Gemini) wsHandleData(respRaw []byte, curr currency.Pair) error {
 	}
 	if _, ok := result["type"]; ok {
 		switch result["type"] {
-		case "subscription_ack":
-			var result WsSubscriptionAcknowledgementResponse
+		case "l2_updates":
+			var l2MarketData *wsL2MarketData
+			err := json.Unmarshal(respRaw, &l2MarketData)
+			if err != nil {
+				return err
+			}
+			return g.wsProcessUpdate(l2MarketData)
+		case "trade":
+			if !g.IsSaveTradeDataEnabled() {
+				return nil
+			}
```

Comment:
Suggestion: I would put this after json.Unmarshal just to alert us if there is a data type change from the exchange side that we don't notice.

Reply:
This is a good point for validation but we already process websocket trades in wsProcessUpdate (they share the same struct) so this is already covered and thus don't have to waste CPU cycles for a second round of validation